### ISSUE TITLE
Is the loss computing right?

### ISSUE THREAD

**Issue by caiohamamuraIFSP** [karpathy/ng-video-lecture#53]
The `logits` are the `idx` transformed by the `token_embedding_table`.

```
logits = self.token_embedding_table(idx)
```

But `targets` are pure `idx`. So when you do:

```
logits = logits.view(B*T, C)
targets = targets.view(B*T)
loss = F.cross_entropy(logits, targets)
```

You are actually calculating the cross_entropy between:
    1. logits: embeddings (probabilities for each token)
    2. targets: indexes for vocabularies

So then you are trying to make embeddings match the target indexes? Does it make sense? Actually I think this is even worse, because you are trying to make the 65 probabilities for each token, match the target indexes.

For example, let's say we have:

```
The
```

And we are analyzing the first letter `T`, which `stoi['T']` is 32. Then when we transform it using the `token_embedding_table`, it will get a probability for each token, that is 65 probabilities (or we can say it is an embedding vector that represents token too). For the seed `1337`, the 32 idx will initially be:

```
tensor([[-0.4895, -0.7685,  0.6316, -0.7093, -0.1722, -0.3041,  0.0524,  1.2048,
         -1.1618,  1.2909,  0.1288,  1.0750, -0.7424, -0.1108, -1.0596, -0.2788,
         -1.7570,  1.3899,  0.6605, -0.2478, -1.6625, -0.5256,  0.0964, -0.2093,
         -0.5770,  1.6780,  0.5273,  0.7738,  0.7493,  1.6907, -0.2628, -1.1123,
         -0.3309,  2.0205,  0.8232, -1.1614, -0.0171, -1.1138,  0.5547,  0.1177,
         -0.2542, -0.1474,  0.0222, -1.1019, -0.2961, -0.4108, -0.4775,  0.0141,
          0.0594, -0.4584,  0.5804,  1.3185, -0.4754,  0.0082, -0.5589,  0.0223,
         -0.6606, -0.1289, -0.0106,  0.1980, -2.6042, -0.0483,  0.9501, -0.7531,
         -0.4020]], device='cuda:0', grad_fn=<EmbeddingBackward0>)
```

But then the actual target was suposed to be `h`, `stoi['h']` is 46.  But if we calculate cross entropy between the embeddings and the `idx` it won't make much sense, but it is even worse, because we are actually comparing the 65 probabilities against the target integer 46, so we are actually computing how far from 46 on average the original tensor will be.

I think we should use:

 - target: transform its idx against `token_embedding_table`
 - prediction: calculate the `logits` `argmax`, and then convert that to the same `token_embedding_table`

Then we are comparing apples against apples, both using the same `token_embedding_table` space.

Or am I missing something?
